### ISSUE TITLE
New Mugen branch is incomplete

### ISSUE THREAD

**Issue by sethcohn** [scherroman/mugen#6]
Went to try the new_mugen code, but it's incomplete still.  ETA on finishing this?

**Comment by scherroman** [comment]
Glad you're eager to try it, so am I 😅 

It's not quite ready to use, but I'm actively working on it and building a test suite alongside. It's getting close. I'd love to get this out within the next month.

**Comment by sethcohn** [comment]
Excellent, btw, I noticed slight build weirdness on both versions. On my Mac, conda seems to not like numpy being installed in the way you do it.  I end up having to manually install numpy in conda directly (not the mugen environment), or I get errors about No such Module (despite that it's there, and won't install again).  Seems like a common issue with numpy, I found complaints all over, but few solutions.  Doing the manual 'conda install numpy' (I think that was the fix, I'll double check when I reinstall the current mugen code, since the new code isn't usable yt)

**Comment by scherroman** [comment]
That's odd...creating the mugen conda environment is just doing a standard pip install for numpy like it does for all the other dependencies...

In any case, once new_mugen is finished and merged into master, i'm going to package and publish it on PyPI so you'll be able to do a simple 'pip install mugen' in any Python environment of your choice. Since moviepy's latest release has the bug-fixes i've been waiting for, and since I've switched to [librosa](https://github.com/librosa/librosa) for beat detection (which is actually available on PyPI), that will give the most basic functionality from mugen out of the box (minus optional tesseract text detection).

My goal is to make mugen both available as a library for use in other python scripts, and also directly from the command line via a command like  'mugen create ...'. Similar to [videogrep](https://github.com/antiboredom/videogrep) and [audiogrep](https://github.com/antiboredom/audiogrep).

**Comment by sethcohn** [comment]
Looking good... installed this on Ubuntu 16.04 by installing Python3.6 using pyenv (I could have used conda too), and then pip installing all of the required items.  Works well so far, looking forward to more.

**Comment by scherroman** [comment]
Closing, as `new_mugen` has been merged into `master` with 49f5b26
